src/engine/cache_engine.c

```diff
@@ -223,36 +223,41 @@ void ocf_seq_cutoff_update(ocf_core_t core, struct ocf_request *req)
 	core->seq_cutoff.bytes += req->byte_length;
 }
 
-ocf_cache_mode_t ocf_get_effective_cache_mode(ocf_cache_t cache,
-		ocf_core_t core, struct ocf_io *io)
+void ocf_resolve_effective_cache_mode(ocf_cache_t cache,
+		ocf_core_t core, struct ocf_request *req)
 {
-	ocf_cache_mode_t mode;
+	ocf_cache_mode_t mode = ocf_cache_mode_none;
 
-	if (cache->pt_unaligned_io && !ocf_req_is_4k(io->addr, io->bytes))
-		return ocf_cache_mode_pt;
+	if (ocf_fallback_pt_is_on(cache)){
```

Comment:
If you move this here, you should exit from the function after checking fallback PT - otherwise the cache mode might get overwritten by global cache mode settings.